utils/utils.js

```diff
@@ -0,0 +1,12 @@
+/**
+ * Adds https:// to a provided URL if it's missing.
+ * @param {string} url The URL string to check against.
+ * @returns The url string with https:// prepended if applicable.
+ */
+const prependProtocol = (url) => {
+  url = decodeURIComponent(url)
+  if (!/^(?:f|ht)tps?\:\/\//.test(url)) {
+    url = "https://" + url
+  }
+  return url
+}
```

Comment:
Wrapped `yumUrl` value sent to the database in this function to ensure the `https://` has been added.